src/hole.rs

```diff
@@ -40,6 +40,38 @@ impl HoleList {
         }
     }
 
+    /// Places a hole of a given size at a given address after a given hole
+    unsafe fn put_hole(hole: &mut Hole, addr: usize, size: usize) {
+        mem::forget(mem::replace(&mut *(addr as *mut Hole),
+                                 Hole {
+                                     size: size,
+                                     next: None,
+                                 }));
+        hole.next = Some(Unique::new(addr as *mut Hole));
+    }
+
+    /// Extends the size of the list by increasing the size of the last hole
+    /// or appending new hole at the end of the list
+    pub fn extend(&mut self, addr: usize, size: usize) {
```

Comment:
Why do we need `extend` and `put_hole`? Can't we just use the `deallocate` function for this?